shared_model/validators/transactions_collection/transactions_collection_validator.hpp

```diff
@@ -27,14 +28,27 @@ namespace shared_model {
               TransactionValidator())
           : transaction_validator_(transactions_validator) {}
 
+      // TODO: IR-1505, igor-egorov, 2018-07-05 Remove method below when
+      // proposal and block will return collection of shared transactions
       /**
        * Validates collection of transactions
        * @param transactions collection of transactions
        * @return Answer containing errors if any
        */
       virtual Answer validate(
           const interface::types::TransactionsForwardCollectionType
-              &transactions) const = 0;
+              &transactions) const {
+        interface::types::SharedTxsCollectionType res;
+        std::for_each(
```

Comment:
I think, it will be better if written as follows:
```
std::transform(std::begin(transactions),
               std::end(transactions),
               std::back_inserter(res),
               [](const auto &tx) { return clone(tx); }); 
```
And also, for the sake of performance, before transforming the space for ```res``` should be reserved, either in vector constructor, or with separate command, such as ```res.reserve(transactions.size());```